### ISSUE TITLE
SIGSEGV with nightly when deserializing object

### ISSUE THREAD

**Issue by epwalsh** [noib3/nvim-oxi#51]
Hey @noib3,

I've ran into an issue where a get a SIGSEGV when calling an exposed function from Nvim nightly that involves deserializing an `Object`. This worked fine on Nvim stable.

Here's a simple module that reproduces the error:

```rust
use nvim_oxi::{self as oxi, object, Dictionary, Function, Object, ObjectKind, Result};
use serde::Deserialize;

#[derive(Default, Deserialize, Debug)]
#[serde(deny_unknown_fields)]
pub struct Config {
    #[serde(default)]
    pub dir: String,
}

fn setup(preferences: Object) -> Result<()> {
    // This works fine as long as preferences is Nil
    let config = match preferences.kind() {
        ObjectKind::Nil => Config::default(),
        _ => {
            // This will cause SIGSEGV
            let deserializer = object::Deserializer::new(preferences);
            Config::deserialize(deserializer)?
        }
    };

    oxi::print!("All good! {config:?}");

    Ok(())
}

#[oxi::module]
fn foobar() -> oxi::Result<Dictionary> {
    Ok(Dictionary::from_iter([(
        "setup",
        Object::from(Function::from_fn(setup)),
    )]))
}
```

From Nvim, calling `:lua require("foobar").setup()` works fine, but `:lua require("foobar").setup({ dir: "./" })` results in

```
'nvim $argv' terminated by signal SIGSEGV (Address boundary error)
```

Again, this worked fine on the latest stable Nvim. Any ideas?

**Comment by noib3** [comment]
Hey thanks for the feedback. I don't think it's related to serde, I can replicate the issue simply with

```rust
use nvim_oxi::{self as oxi, Dictionary, Function, Object, Result};

// Passing a table here causes a segfault.
fn setup(_: Object) -> Result<()> {
    Ok(())
}

#[oxi::module]
fn liboxi() -> Result<Dictionary> {
    Ok(Dictionary::from_iter([("setup", Function::from_fn(setup))]))
}
```

**Comment by noib3** [comment]
This should be fixed in f08099d689a9af5b406965e6b17fa4d07693bce1. There is one gotcha however: an empty table will be deserialized as an array instead of a dictionary:

```lua 
require("foobar").setup({}) -- <- gives error
```

a possible workaround is to implement `Deserialize` for `Config` by hand, implementing the [`Visitor::visit_seq`](https://docs.rs/serde/1.0.142/serde/de/trait.Visitor.html#method.visit_seq) method on your visitor to check if the sequence is empty.

**Comment by epwalsh** [comment]
Thanks @noib3!
